consensus/ibft/ibft.go

```diff
@@ -998,27 +963,18 @@ func (i *Ibft) updateMetrics(block *types.Block) {
 }
 
 func (i *Ibft) insertBlock(block *types.Block) error {
-	committedSeals := make([][]byte, 0)
-
-	for _, commit := range i.state.committed {
-		// no need to check the format of seal here because writeCommittedSeals will check
-		committedSeal, decodeErr := hex.DecodeHex(commit.Seal)
-		if decodeErr != nil {
-			i.logger.Error(
-				fmt.Sprintf(
-					"unable to decode committed seal from %s",
-					commit.From,
-				),
-			)
+	committedSeals := make(map[types.Address][]byte)
 
-			continue
+	for addr, commit := range i.state.committed {
+		committedSeal, err := hex.DecodeHex(commit.Seal)
+		if err != nil {
+			return fmt.Errorf("failed to decode committed seal from %s: %w", addr, err)
```

Comment:
Do we want this to be fatal?

You dropped the `continue` from this conditional. This means that if you have `1` invalid committed seal, you will go into round change